### ISSUE TITLE
object detection dataset annotation

### ISSUE THREAD

**Issue by Tigran050** [ultralytics/ultralytics#21940]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

<img width="1164" height="736" alt="Image" src="https://github.com/user-attachments/assets/34ecf811-6801-4143-9650-fa1edafc2163" />
I am using 850 images with 40000 objects for training YOLO. The data contains only one class. Some images are not well-lit, and I have not annotated them; does this may interfere with the model during the training phase? An example image is attached above. I have another question, the dfl_loss of the model does not drop below 0.750, how can I solve this issue, and how much will it affect the recognition?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Tigran050, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will also assist you here shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
- Your `data.yaml`, model used (e.g., YOLO11 family), `ultralytics` package version, OS, CUDA/PyTorch versions
- Training command and key hyperparameters (img size, epochs, batch, optimizer, augmentations)
- Training logs/plots (loss curves including `dfl_loss`, precision/recall/mAP), and a few sample images with labels
- Label stats or verification outputs and examples of low-light/unannotated images you mentioned
And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Sharing these details will help us provide precise guidance 🙌

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
> Some images are not well-lit, and I have not annotated them; does this may interfere with the model during the training phase?

Yes, you need to label all of them

> I have another question, the dfl_loss of the model does not drop below 0.750, how can I solve this issue, and how much will it affect the recognition?

It doesn't matter if mAP is improving

**Comment by Tigran588** [comment]
> > Some images are not well-lit, and I have not annotated them; does this may interfere with the model during the training phase?
> 
> Yes, you need to label all of them
> 
> > I have another question, the dfl_loss of the model does not drop below 0.750, how can I solve this issue, and how much will it affect the recognition?
> 
> It doesn't matter if mAP is improving

Thanks for the reply. And if the objects are too bright or too dark, will they interfere with the model during training time if I annotate them? I attached the example.

<img width="1153" height="730" alt="Image" src="https://github.com/user-attachments/assets/7d0cb1e4-f3a7-423f-a51b-273e8b7b0ee4" />

**Comment by Y-T-G** [comment]
They will all be trained to be detected if you annotate them all. Otherwise, the labeling would be inconsistent.

**Comment by Tigran588** [comment]
> They will all be trained to be detected if you annotate them all. Otherwise, the labeling would be inconsistent.

<img width="369" height="286" alt="Image" src="https://github.com/user-attachments/assets/ecc67713-cbc1-4d98-b49a-8ee982bc1a45" />

<img width="305" height="237" alt="Image" src="https://github.com/user-attachments/assets/e51b9a2e-2155-4742-9d4f-338a7d347738" />
I have another model that detects and classifies mushroom sorts. I have 1300 images, with equal counts for each class. Each image contains one object. I used a stand to capture the images. When I estimated results on the validation dataset, I had good results. but when I used the model for real-time object detection, I had problems with detection. I think it's because I used the same background for all images.  I have used these augmentation types`
Flip: Horizontal, Vertical
Rotation: Between -15° and +15°
Hue: Between -24° and +24°
Saturation: Between -32% and +32%
Exposure: Between -10% and +10%
Noise: Up to 0.1% of pixels
Bounding Box: Flip: Horizontal, Vertical
Bounding Box: Rotation: Between -15° and +15°

**Comment by Y-T-G** [comment]
If you have similar images with only slight variations, then the model will overfit. You need unique images for a good model.
